### DISCUSSION TITLE
Github Copilot Chat often hallucinating and thinking it's the wrong language

### DISCUSSION THREAD

**Discussion by AlanParadis** [community/community#55819] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

I have a project with a Widget, a CheckBox and a BitmapCheckBox class all in their own files.

While in the BitmapCheckBox cpp file I ask to explain me what it does. It worked really well and it explain me that BitmapCheckBox heritate from CheckBox. Then I ask what's the base class of CheckBox and it give me a wrong answer adding that my code was C# instead of C++.

Conversation :
user : "Explain me how the BitmapCheckBox works"
copilot : "BitmapCheckBox is a subclass of the CheckBox class, [ describe the class... ]"
user : "what's the base class of CheckBox  ?"
copilot : "The base class of CheckBox is Control, which is a base class for all user interface elements in C#" **wrong answer**

Is this a limitation of the actual Github Copilot Chat or is this a bug not finding what I was talking about ? I find it often does understand what I talk about and straight up lie with confidence to me. If it is a limitation why shouldn't it say something like "Sorry this question is out of my scope" ?

**Comment by nk-autoit** [comment]
It's an inherent limitation of large language models. They are generally often right, but many times wrong when it comes to context specific topics.
